File: src/main/java/com/tobe/healthy/common/NotificationSenderInfo.java
package com.tobe.healthy.common;

import com.tobe.healthy.notification.domain.entity.NotificationSenderType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

public class NotificationSenderInfo {

	private NotificationSenderInfo() {
	}

	public static SenderInfo getSenderInfo() {
		return new SenderInfo();
	}

	@Data
	@Builder
	@NoArgsConstructor
	@AllArgsConstructor
	public static class SenderInfo {

		@Builder.Default
		private String profileUrl = "https://cdn.to-be-healthy.shop/origin/profile/default.png";

		@Builder.Default
		private NotificationSenderType senderType = NotificationSenderType.SYSTEM;
	}
}

File: src/main/kotlin/com/tobe/healthy/common/NotificationSenderInfo.kt
package com.tobe.healthy.common

import com.tobe.healthy.notification.domain.entity.NotificationSenderType
import com.tobe.healthy.notification.domain.entity.NotificationSenderType.SYSTEM

object NotificationSenderInfo {
    fun getSenderInfo() : SenderInfo {
        return SenderInfo()
    }

    data class SenderInfo(
        val profileUrl: String = "https://cdn.to-be-healthy.shop/origin/profile/default.png",
        val senderType: NotificationSenderType = SYSTEM
    )
}